dask/dataframe/tests/test_dataframe.py

```diff
@@ -4668,3 +4668,62 @@ def test_dot_nan():
 
     assert_eq(s1.dot(s2), dask_s1.dot(dask_s2))
     assert_eq(s2.dot(df), dask_s2.dot(dask_df))
+
+
+# TODO: Can the setup here go into a fixture? or just a private function?
+def test_pairwise_rejects_right_join():
+    base_df = dd.from_pandas(
+        pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]}, index=[0, 1, 3]), 3
+    )
+    dfs = [
+        dd.from_pandas(
+            pd.DataFrame({"a": [4, 5, 6], "b": [3, 2, 1]}, index=[5, 6, 8]), 3
+        ),
+        dd.from_pandas(
+            pd.DataFrame({"a": [7, 8, 9], "b": [0, 0, 0]}, index=[9, 9, 9]), 3
+        ),
+    ]
+
+    with pytest.raises(ValueError):
+        base_df.join(dfs, how="right")
+
+
+def test_pairwise_rejects_inner_join():
+    base_df = dd.from_pandas(
+        pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]}, index=[0, 1, 3]), 3
+    )
+    dfs = [
+        dd.from_pandas(
+            pd.DataFrame({"a": [4, 5, 6], "b": [3, 2, 1]}, index=[5, 6, 8]), 3
+        ),
+        dd.from_pandas(
+            pd.DataFrame({"a": [7, 8, 9], "b": [0, 0, 0]}, index=[9, 9, 9]), 3
+        ),
+    ]
+
+    with pytest.raises(ValueError):
+        base_df.join(dfs, how="inner")
+
+
+def test_pairwise_merge_results_in_identical_output_df():
+    dfs_to_merge = []
+    ddf_loop = dd.from_pandas(pd.DataFrame(index=[0, 1, 2, 3]), 3)
+    for i in range(10):
+        df = pd.DataFrame(
+            {
+                f"{i}A": [4, 5, 6],
+                f"{i}B": [3,2,1],
+            },
+            index=[0, 1, 3],
+        )
+        ddf = dd.from_pandas(df, 3)
+        dfs_to_merge.append(ddf)
+
+    for ddf in dfs_to_merge:
+        ddf_loop = ddf_loop.join(ddf, how="left")
+
+    ddf_pairwise = dd.from_pandas(pd.DataFrame(index=[0, 1, 2, 3]), 3)
+
+    ddf_pairwise = ddf_pairwise.join(dfs_to_merge)
+
+    tm.assert_frame_equal(ddf_pairwise.compute(), ddf_loop.compute())
```

Comment:
Use the dask.dataframe `assert_eq` function for a full check of the dask metdata as well as the computed values.

```suggestion
    assert_eq(ddf_pairwise, ddf_loop)
```